image_services/rtc_services/rgb/make_rgb_service.py

```diff
@@ -0,0 +1,220 @@
+)
+
+# Set mosaic dataset properties
+
+print(f'Setting properties for {mosaic_dataset}...')
+arcpy.management.SetMosaicDatasetProperties(
+    in_mosaic_dataset=mosaic_dataset,
+    rows_maximum_imagesize=5000,
+    columns_maximum_imagesize=5000,
+    allowed_compressions='JPEG;NONE;LZ77',
+    default_compression_type='JPEG',
+    JPEG_quality=80,
+    resampling_type='NEAREST',
+    LERC_Tolerance=0.01,
+    clip_to_footprints='CLIP',
+    clip_to_boundary='CLIP',
+    color_correction='NOT_APPLY',
+    footprints_may_contain_nodata='FOOTPRINTS_MAY_CONTAIN_NODATA',
+    allowed_mensuration_capabilities='BASIC',
+    default_mensuration_capabilities='BASIC',
+    allowed_mosaic_methods='Center;NorthWest;Nadir;LockRaster;ByAttribute;Seamline;None',
+    default_mosaic_method='ByAttribute',
+    order_field='StartDate',
+    order_base='1/1/2050 12:00:00 AM',
+    sorting_order='Ascending',
+    mosaic_operator='FIRST',
+    blend_width=10,
+    view_point_x=300,
+    view_point_y=300,
+    max_num_per_mosaic=50,
+    cell_size_tolerance=1.8,
+    cell_size=3,
+    metadata_level='BASIC',
+    transmission_fields='Name;StartDate;EndDate;MinPS;MaxPS;LowPS;HighPS;Date;ZOrder;Dataset_ID;CenterX;CenterY;Tag;ProductName;GroupName;DownloadURL',
+    use_time='ENABLED',
+    start_time_field='StartDate',
+    end_time_field='EndDate',
+    max_num_of_download_items=50,
+    max_num_of_records_returned=2000,
+    processing_templates=f'{raster_function_template};None',
+    default_processing_template=raster_function_template,
+)
+
+print('Calculating cell size ranges...')
+arcpy.management.CalculateCellSizeRanges(
+    in_mosaic_dataset=mosaic_dataset,
+    do_compute_min='NO_MIN_CELL_SIZES',
+    do_compute_max='NO_MAX_CELL_SIZES',
+    max_range_factor=10,
+    cell_size_tolerance_factor=0.8,
+    update_missing_only='UPDATE_ALL',
+)
+
+# Generate mosaic dataset overview and add it to the mosaic dataset
+
+print(f'Generating {local_overview}...')
+with arcpy.EnvManager(cellSize=1200):
+    arcpy.management.CopyRaster(
+        in_raster=mosaic_dataset,
+        out_rasterdataset=local_overview,
+    )
+
+print(f'Moving CRF to {s3_overview}...')
+subprocess.run(['aws', 's3', 'cp', local_overview, s3_overview.replace("/vsis3/", "s3://"), '--recursive'])
+
+print('Adding overview to mosaic dataset...')
+arcpy.management.AddRastersToMosaicDataset(
+    in_mosaic_dataset=mosaic_dataset,
+    raster_type='Raster Dataset',
+    input_path=s3_overview,
+)
+
+# Calculate the date range of the items in the mosaic dataset
+ds_cursor = arcpy.da.SearchCursor(mosaic_dataset, ["Tag", "StartDate"])
+stdatelist = []
+if (ds_cursor is not None):
+    print('Determining Start and End Dates...')
+    for row in ds_cursor:
+        if row[0] != 'Dataset':
+            stdatelist.append(row[1])
+    stdate = min(stdatelist)
+    endate = max(stdatelist)
+
+stdate_buffer = stdate + datetime.timedelta(hours=-8)
+endate_buffer = endate + datetime.timedelta(hours=8)
```

Comment:
If `ds_cursor` is `None`, then `stdate` and `endate` will be undefined here. I'm not sure what behavior we want here. In what cases will `ds_cursor` be `None`?

Reply:
Good point. I guess the only reason the ds_cursor would be none is if it either can't find the mosaic dataset, or the mosaic dataset is missing either the right fields or doesn't have any records. It was probably included in the original script to allow processing to continue, but I'm not sure if that's what would happen here. It's true that if we can't calculate the start and end dates, we also wouldn't want to calculate the buffered start and end dates, but this may be a failsafe that is somewhat irrelevant. Maybe we WOULD want the script to error out if we can't do this, because it indicates something problematic. I'm okay with paring this down.

Reply:
I've replaced the if statement for the ds_cursor with a with statement. It's possible that the if statement was serving to close the search cursor after the process was complete, and the with statement should also serve that function.